### PR TITLE
Administrator selection for access request approval

### PR DESCRIPTION
**Pull request by leonardlorenz**
Introduces:
- allow for (optional) cluster access requests to specific administrators instead of all of them
- new test for an optional administrator parameter to User.send_access_request()
- changed /access/request/ route from GET to POST to allow for form data passing and rewrote the according tests

(Implements #68 )

### CODE REVIEW

**Comment by leonardlorenz** [comment]
Please have a closer look at line 212 in models.py. I'm not sure if this generates a lot of queries or not.
When testing the e-mails also never arrived even though send_mail() reported they did. Is there a reason for this that I don't know yet? When testing I vpn into the universities network, maybe it is related to that.

File: kubeportal/templates/portal_welcome.html [hunk 1]
```diff
@@ -21,9 +21,34 @@
           Sorry, you have no access to the cluster.
         </p>
         {% if user.has_access_requested %}
```
Comment (troeger) [inline kubeportal/templates/portal_welcome.html:23]: I only see a difference in the text, the form is the same. Please optimize that.

File: kubeportal/models.py [hunk 2]
```diff
@@ -176,12 +176,15 @@ def has_access_requested(self):
         return result
 
     @transition(field=state, source=[UserState.NEW, UserState.ACCESS_REQUESTED, UserState.ACCESS_APPROVED, UserState.ACCESS_REJECTED], target=UserState.ACCESS_REQUESTED)
-    def send_access_request(self, request):
+    def send_access_request(self, request, *args, **kwargs):
```
Comment (troeger) [inline kubeportal/models.py:179]: This looks like overkill. Wouldn't it be good enough to have a new parameter "administrator=None" for the method?. The default ("None") stands for sending it to all, setting it chooses one.

File: kubeportal/models.py [hunk 3]
```diff
@@ -194,12 +197,22 @@ def send_access_request(self, request):
...
+
+        if administrator:
+            cluster_admins.append(User.objects.get(username=administrator))
+            logger.info(f"Sending access request from '{self.username}' to '{administrator}'")
+        else:
+            for admin in User.objects.filter(is_superuser=True):
+                cluster_admins.append(admin)
+            logger.info(f"Sending access request from '{self.username}' to all administrators")
+
+        cluster_admin_emails = [admin.email for admin in cluster_admins]
```
Comment (leonardlorenz) [inline kubeportal/models.py:210]: I was referring to this when I meant line 212. Next time I'll use comments first.

**Review by troeger — APPROVED** [review]
Perfect, thanks.

**Comment by troeger** [comment]
Please fix the merge conflicts, after that we are ready to get this into master.

**Review by troeger — APPROVED** [review]

**Pull request merged**